pkg/api/bytes.go

```diff
@@ -115,6 +115,7 @@ func (s *Service) bytesUploadHandler(w http.ResponseWriter, r *http.Request) {
 		jsonhttp.InternalServerError(w, "sync chunks failed")
 		return
 	}
+	timer.Stop()
```

Comment:
Would we set timeout on reading chunks using this construct:
```
pctx, cancel := context.WithTimeout(ctx, time.Minute*5)
address, err := p(pctx, r.Body)
```

Reply:
`context.WithTimeout` is what is generally used across codebase, as long as error is logged it will be sufficient information to know what has happened (because error message would state `context deadline exceeded`).

Reply:
> `context.WithTimeout` is what is generally used across codebase, as long as error is logged it will be sufficient information to know what has happened (because error message would state `context deadline exceeded`).

This is not the main reason, as already mentioned, it is better to limit only the problematic part for the time being.

Reply:
if 5min is not enough for entire request, we will see it monitoring and we could decided to rise timeout a bit more.

simpler solution is what's going to be more effective.

Reply:
> the part of request currently monitored with timer is big majority of entire request execution time (other part of request is neglectable compared to whole).

Have you done some measurements on this or is that just an assumption?

> simpler solution is what's going to be more effective.

Effective in what way?

Reply:
this PR is simplifying the logic which was previously implemented with `TimeoutReader`. 
following same idea of simplification, this comment proposed to start with even simpler change, which is to utilize `context.WithTimeout`.

why to use `context.WithTimeout` instead of timer?
it appears that if context, created with timeout, passed to `p(pctx, r.Body)` would provide almost identical effect as timer.
idea would be to use simplest approach possible first, since it appears that it could be sufficient enough.